src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -43,6 +46,145 @@
+        final List<CustomerInvoiceDetail> awardAccountingLines = new ArrayList<>();
+        if (!CollectionUtils.isEmpty(contractsGrantsInvoiceDocument.getAccountDetails())) {
+            final Map<String, KualiDecimal> accountExpenditureAmounts = getCategoryExpenditureAmountsForInvoiceAccountDetail(
+                    contractsGrantsInvoiceDocument);
+            final Map<String, KualiDecimal> accountTotalBilledAmounts = getCategoryTotalBilledAmountsForInvoiceAccountDetail(
+                    contractsGrantsInvoiceDocument);
+            for (final InvoiceAccountDetail invAcctD : contractsGrantsInvoiceDocument.getAccountDetails()) {
+                final String proposalNumber = invAcctD.getProposalNumber();
+                final String chartOfAccountsCode = invAcctD.getChartOfAccountsCode();
+                final String accountNumber = invAcctD.getAccountNumber();
+                if (invAcctD.getAccount() == null) {
+                    invAcctD.refreshReferenceObject(KFSPropertyConstants.ACCOUNT);
+                }
+                final SubFundGroup subFundGroup = invAcctD.getAccount().getSubFundGroup();
+                final Integer sequenceNumber = contractsGrantsInvoiceDocument.getAccountDetails().indexOf(invAcctD) + 1;
+
+                final String accountKey = StringUtils.join(new String[]{chartOfAccountsCode, accountNumber}, "-");
+                KualiDecimal totalAmount = accountExpenditureAmounts.getOrDefault(accountKey, KualiDecimal.ZERO);
+
+                if (invAcctD.getTotalPreviouslyBilled().isZero()) {
+                    KualiDecimal previouslyBilledAmount = getPredeterminedBillingBilledToDateAmount(proposalNumber,
+                            chartOfAccountsCode, accountNumber);
+                    previouslyBilledAmount = previouslyBilledAmount.add(
+                            getMilestonesBilledToDateAmount(proposalNumber, chartOfAccountsCode, accountNumber));
+
+                    final KualiDecimal totalBilledAmount = accountTotalBilledAmounts.getOrDefault(accountKey,
+                            KualiDecimal.ZERO);
+
+                    previouslyBilledAmount = previouslyBilledAmount.subtract(totalBilledAmount);
+                    if (previouslyBilledAmount.isGreaterThan(KualiDecimal.ZERO)) {
+                        totalAmount = totalAmount.subtract(previouslyBilledAmount);
+                    }
+                }
+
+                final CustomerInvoiceDetail cide = createSourceAccountingLineBackport(contractsGrantsInvoiceDocument,
+                        chartOfAccountsCode,
+                        accountNumber,
+                        subFundGroup,
+                        totalAmount,
+                        sequenceNumber
+                );
+                awardAccountingLines.add(cide);
+            }
+        }
+        return awardAccountingLines;
+    }
+
+
+    protected CustomerInvoiceDetail createSourceAccountingLineBackport(
```

Comment:
why did you need to rename createSourceAccountingLine to createSourceAccountingLineBackport?
I think you can use the same function name as in basecode, and include an override annotation

Reply:
I think this aspect is OK, I am fine with this